### PR TITLE
DISTMYSQL-440: Orchestrator should use the new source-replica terminology for any higher version than 8.3

### PR DESCRIPTION
**Pull request by kamil-holubicki**
https://perconadev.atlassian.net/browse/DISTMYSQL-440

Problem 1:
Orchestrator uses source-replica queries only for 8.4.x versions.
For versions like 9.x it uses old master-slave queries.

Reason:
Query String Provider detects only 8.4.x versions as "new ones".

Solution:
Change "new version" detection method.

Problem 2:
MySQL 9.x removed native authentication plugin.
Now caching_sha2_password is used, and requres either secure SSL
connection or at least RSA-based password exchange during the client
authentication. If replication uses SSL channels, and we move the source
to be replica, we need to enable SSL on it. We do it by executing
'change replication source to source_ssl=1' if the new source has SSL
enabled.
However, if replication does not use SSL it means we used RSA, so after
making the source to be replica, we need to enable RSA-based login on it
by executing 'change replication source to get_source_public_key=1'.

Additionally, updated dbdeployer to be compatible with 9.0 (replication
user does not use native authentication)

### CODE REVIEW

**Comment by kamil-holubicki** [comment]
https://ps80.cd.percona.com/job/mysql-orchestrator-pipeline/63

File: go/inst/instance_topology_dao.go [hunk 1]
```diff
@@ -871,6 +871,32 @@ func ChangeMasterCredentials(instanceKey *InstanceKey, creds *ReplicationCredent
...
+		return instance, log.Errore(err)
+	}
+
+	if instance.ReplicationThreadsExist() && !instance.ReplicationThreadsStopped() {
+		return instance, fmt.Errorf("EnableMasterGetSourcePublicKey: Cannot enable GetSourcePublicKey replication on %+v because replication threads are not stopped", *instanceKey)
+	}
+	log.Debugf("EnableMasterSSL: Will attempt enabling GetSourcePublicKey replication on %+v", *instanceKey)
+
+	if *config.RuntimeCLIFlags.Noop {
+		return instance, fmt.Errorf("noop: aborting CHANGE MASTER TO GET_SOURCE_PUBLIC_KEY=1 operation on %+v; signaling error but nothing went wrong.", *instanceKey)
```
Comment (venkatesh-prasad-v) [inline go/inst/instance_topology_dao.go:886]:
```suggestion
		return instance, fmt.Errorf("noop: aborting CHANGE REPLICATION SOURCE TO GET_SOURCE_PUBLIC_KEY=1 operation on %+v; signaling error but nothing went wrong.", *instanceKey)
```

File: go/inst/instance_topology_dao.go [hunk 2]
```diff
@@ -871,6 +871,32 @@ func ChangeMasterCredentials(instanceKey *InstanceKey, creds *ReplicationCredent
...
+	}
+	_, err = ExecInstance(instanceKey, instance.QSP.change_master_to_get_source_public_key())
+
+	if err != nil {
+		return instance, log.Errore(err)
+	}
+
+	log.Infof("EnableMasterSSL: Enabled GetSourcePublicKey replication on %+v", *instanceKey)
```
Comment (venkatesh-prasad-v) [inline go/inst/instance_topology_dao.go:894]:
```suggestion
	log.Infof("EnableMasterGetSourcePublicKey: Enabled GetSourcePublicKey replication on %+v", *instanceKey)
```

Comment (kamil-holubicki) [reply go/inst/instance_topology_dao.go]: done

File: go/inst/instance_topology_dao.go (hunk 1 above)
Comment (kamil-holubicki) [reply go/inst/instance_topology_dao.go]: done

**Comment by kamil-holubicki** [comment]
https://ps80.cd.percona.com/job/mysql-orchestrator-pipeline/78/

**Review by venkatesh-prasad-v — APPROVED** [review]

**Pull request merged**